dashboard/views.py

```diff
@@ -143,6 +145,86 @@ def post(self, request):
         )
 
 
+class UserExamEnrollment(APIView):
+    """
+    Create an exam enrollment for the user in a given exam run identified by course_id.
+    """
+    authentication_classes = (
+        authentication.SessionAuthentication,
+        authentication.TokenAuthentication,
+    )
+    permission_classes = (permissions.IsAuthenticated, )
+
+    def post(self, request):
```

Comment:
This is a bit of a lengthy implementation for a view, if there's a way to push some of this into an api function and then do the error response handling in the view still that'd be a little cleaner.

Reply:
I tried taking some things out, but the majority of the view is just error messages. Let me know if you think, we can cut something out.